pkg/reconciler/knativeserving/ingress/istio.go

```diff
@@ -18,23 +18,36 @@ package ingress
 
 import (
 	"context"
+	"strconv"
 
 	mf "github.com/manifestival/manifestival"
 	"go.uber.org/zap"
 	istionetworkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
 	"istio.io/client-go/pkg/clientset/versioned/scheme"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"knative.dev/operator/pkg/apis/operator/base"
 	"knative.dev/operator/pkg/apis/operator/v1beta1"
 	servingv1beta1 "knative.dev/operator/pkg/apis/operator/v1beta1"
 	"knative.dev/pkg/logging"
 )
 
+const enableSecretInformerFilteringByCertUIDAnno = "knative.dev/enable-secret-informer-filtering-by-cert-uid"
```

Comment:
Kubernetes operator's custom resources are able to configure the operands, by leverage the CR's spec. It is a bit strange to use the label to configure.

I suggest not bother the fields unless they are under `spec`.
You can add a boolean flag in the struct `type IstioIngressConfiguration struct` and change the serving CRD according. Let's configure everything with the CR's spec.

Reply:
It is an annotation and it common practice to have such extra config info passed via annotations. I thought about that but it looked a bit intrusive to add in the CR since this looks an internal feature and it might be always on at some point.